app/db_version.go

```diff
@@ -0,0 +1,46 @@
+package app
+
+import (
+	"os"
+	"path"
+	"strconv"
+
+	"github.com/pkg/errors"
+)
+
+const currentDatabaseVersion = 1
+
+func checkDatabaseVersion(dbPath string) error {
+	dbVersionFileName := path.Join(dbPath, "version")
+	versionBytes, err := os.ReadFile(dbVersionFileName)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return createDatabaseVersionFile(dbVersionFileName)
+		}
+		return err
+	}
+
+	databaseVersion, err := strconv.Atoi(string(versionBytes))
+	if err != nil {
+		return err
+	}
+
+	if databaseVersion != currentDatabaseVersion {
```

Comment:
Maybe change to if `databaseVersion > currentDatabaseVersion`? it's just easier IMHO

Reply:
I disagree, we are going to support some upgrade paths, but not necessarily all of them.
Threrefore, I'd rather be as prohibitive as possible, and then whomever writes the upgrade logic will be able to permit the specific things he wishes to permit.